File: utils-io/src/main/java/org/wdt/utils/io/FileUtils.java
package org.wdt.utils.io;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.nio.file.Files;
import java.nio.file.attribute.FileTime;
import java.util.Date;
import java.util.List;

public class FileUtils {
    public static final File HERE = new File(System.getProperty("user.dir"));
    public static String readFileToString(File file) throws IOException {
        return PathUtils.readFileToString(file.toPath());
    }

    public static void writeStringToFile(File file, String s, boolean append) throws IOException {
        PathUtils.writeStringToFile(file.toPath(), s, append);
    }

    public static void writeStringToFile(File file, String s) throws IOException {
        writeStringToFile(file, s, false);
    }


    public static void touch(File file) throws IOException {
        PathUtils.touch(file.toPath());
    }

    public static List<String> readFileToLines(File file) throws IOException {
        return PathUtils.readFileToLines(file.toPath());
    }

    public static void delete(File file) throws IOException {
        PathUtils.delete(file.toPath());
    }

    public static void cleanDirectory(File file) throws IOException {
        ckeckIsDirectory(file);
        PathUtils.cleanDirectory(file.toPath());
    }


    public static long sizeOfDirectory(File file) throws IOException {
        ckeckIsDirectory(file);
        return PathUtils.sizeOfDirectory(file.toPath());
    }

    public static void deleteDirectory(File file) throws IOException {
        ckeckIsDirectory(file);
        PathUtils.deleteDirectory(file.toPath());
    }

    public static void copyFile(File srcFile, File file) throws IOException {
        Files.copy(Files.newInputStream(srcFile.toPath()), file.toPath());
    }


    public static InputStream newInputStream(File file) throws IOException {
        return Files.newInputStream(file.toPath());
    }

    public static OutputStream newOutputStream(File file) throws IOException {
        return Files.newOutputStream(file.toPath());
    }

    public static void createDirectories(File file) throws IOException {
        Files.createDirectories(file.toPath());
    }

    public static String getCanonicalPath(File file) {
        try {
            return file.getCanonicalPath();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public static boolean isFileOlder(File file, Date date) throws IOException {
        return isFileOlder(file, date.getTime());
    }

    public static boolean isFileOlder(File file, long l) throws IOException {
        return Files.getLastModifiedTime(file.toPath()).compareTo(FileTime.fromMillis(l)) < 0;
    }


    public static void ckeckIsDirectory(File file) throws IOException {
        if (!file.isDirectory()) {
            throw new IOException("This File isn't Directory");
        }
    }

    public static boolean isFileNotExists(File file) {
        return !file.exists();
    }

    public static boolean isFileExists(File file) {
        return file.exists();
    }

    public static boolean isFileNotExistsAndIsNotSameSize(File file, long l) throws IOException {
        return PathUtils.isFileNotExistsAndIsNotSameSize(file.toPath(), l);
    }

    public static File toFile(String s) {
        return new File(s);
    }

    public static File toFile(String parent, String child) {
        return new File(toFile(parent), child);
    }

    public static File toFile(File parent, String child) {
        return new File(parent, child);
    }

    public static File toFile(File parent, String name, String extension) {
        return new File(parent, name + "." + extension);
    }

    public static String getFileSha1(File file) throws IOException {
        return IOUtils.getInputStreamSha1(newInputStream(file));
    }

    public static long sizeOf(File file) throws IOException {
        return PathUtils.sizeOf(file.toPath());
    }

    public static void deleteFile(File file) throws IOException {
        if (isFileNotExists(file)) {
            return;
        }
        PathUtils.deleteFile(file.toPath());
    }
}

File: utils-io/src/main/kotlin/org/wdt/utils/io/FileUtils.kt
package org.wdt.utils.io

import java.io.File
import java.io.FileReader
import java.io.InputStream
import java.io.OutputStream
import java.util.*

fun File.readFileToString(): String {
    return FileUtils.readFileToString(this)
}


fun File.writeStringToFile(string: String, boolean: Boolean = false) {
    FileUtils.writeStringToFile(this, string, boolean)
}

fun File.touch() {
    FileUtils.touch(this)
}

fun File.readFileToLine(): List<String> {
    return FileUtils.readFileToLines(this)
}

fun File.sizeOf(): Long {
    return PathUtils.sizeOf(this.toPath())
}

fun File.deleteDirectory() {
    FileUtils.deleteDirectory(this)
}

fun File.newOutputStream(): OutputStream {
    return FileUtils.newOutputStream(this)
}

fun File.newInputStream(): InputStream {
    return FileUtils.newInputStream(this)
}

fun File.isFileOlder(date: Date): Boolean {
    return FileUtils.isFileOlder(this, date)
}

fun File.isFileOlder(long: Long): Boolean {
    return FileUtils.isFileOlder(this, long)
}

fun File.createDirectories() {
    FileUtils.createDirectories(this)
}

fun File.copyFile(file: File) {
    FileUtils.copyFile(this, file)
}

fun File.sizeOfDirectory(): Long {
    return FileUtils.sizeOfDirectory(this)
}

fun File.isFileNotExistsAndIsNotSameSize(long: Long): Boolean {
    return FileUtils.isFileNotExistsAndIsNotSameSize(this, long)
}

fun File.isFileExists(): Boolean {
    return FileUtils.isFileExists(this)
}

fun File.isFileNotExists(): Boolean {
    return FileUtils.isFileNotExists(this)
}

fun File.getExtension(): String {
    return this.extension
}

fun File.getFileSha1(): String {
    return FileUtils.getFileSha1(this)
}

fun String.toFile(): File {
    return File(this)
}

fun String.toFile(child: String): File {
    return File(this, child)
}

fun File.toFile(child: String): File {
    return File(this, child)
}

fun File.newReader(): FileReader {
    return FileReader(this)
}